langchain/prompts/few_shot.py

```diff
@@ -120,8 +140,73 @@ def _prompt_type(self) -> str:
         """Return the prompt type key."""
         return "few_shot"
 
-    def dict(self, **kwargs: Any) -> Dict:
-        """Return a dictionary of the prompt."""
-        if self.example_selector:
-            raise ValueError("Saving an example selector is not currently supported")
-        return super().dict(**kwargs)
+
+# For backwards compatibility.
+FewShotPromptTemplate = FewShotStringPromptTemplate
+
+
+class FewShotChatPromptTemplate(BaseChatPromptTemplate, BaseFewShotPromptTemplate):
```

Comment:
i think more generically, the suffix and prefix here should be ChatPromptTemplates (so there can be multiple messages)

Reply:
Good point